src/core_ocean/shared/mpas_ocn_equation_of_state_wright.F

```diff
@@ -0,0 +1,717 @@
+!> \author  Xylar Asay-Davis
+!> \date    July 8, 2019
+!
+!-----------------------------------------------------------------------
+
+   subroutine compute_boussinesq_p(nVertLevels, nCells, kDisplaced,      &
+                                   displacementType, zMid, maxLevelCell, &
+                                   boussinesqPres)
+   !{{{
+   !--------------------------------------------------------------------
+
+      !-----------------------------------------------------------------
+      ! Input variables
+      !-----------------------------------------------------------------
+
+      integer, intent(in) :: &
+         nCells,             &! number of horizontal cells
+         nVertLevels,        &! max number of verical levels
+         kDisplaced           ! target layer for displacement
+
+      character(len=*), intent(in) :: &
+         displacementType     ! choice of displacement
+
+      real (kind=RKIND), dimension(:,:), intent(in) :: &
+         zMid
+
+      integer, dimension(:), intent(in) :: &
+         maxLevelCell
+
+      !-----------------------------------------------------------------
+      ! Output variables
+      !-----------------------------------------------------------------
+
+      real (kind=RKIND), dimension(:,:), intent(out) :: &
+         boussinesqPres
+
+      !-----------------------------------------------------------------
+      ! Local variables
+      !-----------------------------------------------------------------
+
+      integer ::          &
+         iCell, k,        &! cell and vertical level loop indices
+         kPressure         ! index to determine ref level for pressure
+
+      !-----------------------------------------------------------------
+      !  if displacementType = 'relative', density is calculated
+      !     with pressure referenced to level k + kDisplaced
+      !     If kDisplaced=0, in-situ density is returned (no displacement)
+      !     If kDisplaced/=0, a displaced density is returned
+      !
+      !  if displacementType = 'absolute', potential density is calculated
+      !     referenced to zero pressure
+
+      if ((displacementType == 'relative') .or. &
+          (displacementType == 'surfaceDisplaced')) then
+
+         !$omp parallel
+         !$omp do schedule(runtime) private(k, kPressure)
+         do iCell=1,nCells
+            do k = 1, nVertLevels
+               kPressure = min(k + kDisplaced, maxLevelCell(iCell))
+               kPressure = max(kPressure, 1)
+               boussinesqPres(k, iCell) = -rho_sw*gravity*zMid(kPressure, iCell)
+            end do
+         end do
+         !$omp end do
+         !$omp end parallel
+
+      else ! displacementType == 'absolute'
+
+         !$omp parallel
+         !$omp do schedule(runtime) private(k)
+         do iCell=1,nCells
+            do k = 1, nVertLevels
+               boussinesqPres(k, iCell) = config_eos_wright_ref_pressure
```

Comment:
@cbegeman and @dengwirda, for potential density, the pressure is, instead, a constant reference value (0.0 by default).